crypto/pkcs12/pbkdf.go

```diff
@@ -0,0 +1,214 @@
+// Entire File adapted from https://github.com/Azure/go-pkcs12/blob/master/pbkdf.go
```

Comment:
This file could be its own package. The reference states that is comes from section 6.1.2 of https://tools.ietf.org/html/rfc2898, which could be added here for clarity.